imagersite/imager_images/tests/test_views.py

```diff
@@ -0,0 +1,115 @@
+from django.contrib.auth.models import User
+from django.test import TestCase, RequestFactory
+from django.urls import reverse, reverse_lazy
+from django.http import Http404
+from ..views import library_view, photo_view
+from ..models import Album, Photo
+from model_mommy import mommy
+
+
+class ViewTests(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        """Set up some users, albums and photos."""
+        super(TestCase, cls)
+        for _ in range(5):
+            user = mommy.make(User, password='CodeFellows')
+            photo = mommy.make(Photo, user=user)
+            mommy.make(Album, user=user, cover=photo)
+
+    @classmethod
+    def tearDownClass(cls):
+        """tears down test objects"""
+        super(TestCase, cls)
+        Photo.objects.all().delete()
+        Album.objects.all().delete()
+
+    def test_photo_view(self):
+        """Validate photo view exists and renders."""
+        photo = Photo.objects.all().first()
+        photo.user = User.objects.all().first()
+        photo.published = 'PUBLIC'
+        photo.save()
+        response = self.client.get(reverse('photo', args=[photo.id]))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'imager_images/photo.html')
+
+    def test_non_public_photo_raises_404(self):
+        """Validate a non-public photo raises a 404."""
+        user = User.objects.first()
+        photo = Photo.objects.exclude(user=user).first()
+        photo.published = 'PRIVATE'
+        photo.save()
+        request = RequestFactory().get(reverse_lazy('photo', args=[photo.id]))
```

Comment:
Just so you know, the argument passed into this `.get()` does not matter. It can be any string because you are passing the `request` directly into the view function.